// @vue-email/font/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  Font: () => Font
});
module.exports = __toCommonJS(src_exports);
var import_vue = require("vue");
var Font = (0, import_vue.defineComponent)({
  name: "Font",
  props: {
    fontFamily: {
      type: String,
      required: true
    },
    fallbackFontFamily: {
      type: [String, Array],
      default: "Arial"
    },
    webFont: {
      type: Object,
      default: void 0
    },
    fontStyle: {
      type: String,
      default: "normal"
    },
    fontWeight: {
      type: [String, Number],
      default: 400
    }
  },
  setup({ fontFamily, fallbackFontFamily, webFont, fontStyle, fontWeight }) {
    const src = webFont ? `src: url(${webFont.url}) format('${webFont.format}');` : "";
    const style = `
      @font-face {
        font-family: '${fontFamily}';
        font-style: ${fontStyle};
        font-weight: ${fontWeight};
        mso-font-alt: '${Array.isArray(fallbackFontFamily) ? fallbackFontFamily[0] : fallbackFontFamily}';
        ${src}
      }
  
      * {
        font-family: '${fontFamily}', ${Array.isArray(fallbackFontFamily) ? fallbackFontFamily.join(", ") : fallbackFontFamily};
      }
    `;
    return () => {
      return (0, import_vue.h)("style", {
        innerHTML: style
      });
    };
  }
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Font
});
