// @vue-email/preview/dist/index.mjs
// src/index.ts
import { computed, defineComponent, h } from "vue";
var PREVIEW_MAX_LENGTH = 150;
var renderWhiteSpace = (text) => {
  if (text.length >= PREVIEW_MAX_LENGTH) {
    return null;
  }
  const whiteSpaceCodes = "\xA0\u200C\u200B\u200D\u200E\u200F\uFEFF";
  return h("div", [whiteSpaceCodes.repeat(PREVIEW_MAX_LENGTH - text.length)]);
};
var Preview = defineComponent({
  name: "Preview",
  setup(_, { slots }) {
    const text = computed(() => {
      if (slots.default !== void 0) {
        const children = slots.default()[0].children;
        const newText = Array.isArray(children) ? children.join("") : children;
        return newText == null ? void 0 : newText.substring(0, PREVIEW_MAX_LENGTH);
      }
      return "";
    });
    return () => {
      return h(
        "div",
        {
          style: {
            display: "none",
            overflow: "hidden",
            lineHeight: "1px",
            opacity: 0,
            maxHeight: 0,
            maxWidth: 0
          }
        },
        [text.value, renderWhiteSpace(text.value)]
      );
    };
  }
});
export {
  Preview,
  renderWhiteSpace
};
