// @vue-email/link/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  Link: () => Link
});
module.exports = __toCommonJS(src_exports);
var import_vue = require("vue");
var Link = (0, import_vue.defineComponent)({
  name: "Link",
  props: {
    href: {
      type: String,
      required: true
    },
    target: {
      type: String,
      default: "_blank"
    }
  },
  setup(props, { slots }) {
    return () => {
      var _a;
      return (0, import_vue.h)(
        "a",
        {
          "style": "color: #067df7; text-decoration: none",
          "href": props.href,
          "target": props.target
        },
        (_a = slots.default) == null ? void 0 : _a.call(slots)
      );
    };
  }
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Link
});
