// @vue-email/img/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  Img: () => Img
});
module.exports = __toCommonJS(src_exports);
var import_vue = require("vue");
var Img = (0, import_vue.defineComponent)({
  name: "Img",
  props: {
    src: {
      type: String,
      required: true
    },
    width: {
      type: [String, Number],
      required: false
    },
    height: {
      type: [String, Number],
      required: false
    },
    alt: {
      type: String,
      required: false
    }
  },
  setup(props) {
    const src = (0, import_vue.ref)(props.src);
    return () => {
      return (0, import_vue.h)("img", {
        "style": "display: block; outline: none; border: none; text-decoration: none",
        src: src.value,
        width: props.width,
        height: props.height,
        alt: props.alt
      });
    };
  }
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Img
});
