// @vue-email/heading/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  Heading: () => Heading
});
module.exports = __toCommonJS(src_exports);
var import_vue = require("vue");

// src/utils/spaces.ts
var withMargin = (props) => {
  const nonEmptyStyles = [
    withSpace(props.m, ["margin"]),
    withSpace(props.mx, ["marginLeft", "marginRight"]),
    withSpace(props.my, ["marginTop", "marginBottom"]),
    withSpace(props.mt, ["marginTop"]),
    withSpace(props.mr, ["marginRight"]),
    withSpace(props.mb, ["marginBottom"]),
    withSpace(props.ml, ["marginLeft"])
  ].filter((s) => Object.keys(s).length);
  const mergedStyles = nonEmptyStyles.reduce((acc, style) => {
    return { ...acc, ...style };
  }, {});
  return mergedStyles;
};
var withSpace = (value, properties) => {
  return properties.reduce((styles, property) => {
    if (!isNaN(parseFloat(value))) {
      return { ...styles, [property]: `${value}px` };
    }
    return styles;
  }, {});
};

// src/index.ts
var Heading = (0, import_vue.defineComponent)({
  name: "Heading",
  props: {
    as: {
      type: String,
      default: "h1"
    },
    m: {
      type: [String, Number],
      default: void 0
    },
    mx: {
      type: [String, Number],
      default: void 0
    },
    my: {
      type: [String, Number],
      default: void 0
    },
    mt: {
      type: [String, Number],
      default: void 0
    },
    mr: {
      type: [String, Number],
      default: void 0
    },
    mb: {
      type: [String, Number],
      default: void 0
    },
    ml: {
      type: [String, Number],
      default: void 0
    },
    style: {
      type: [Object, String],
      default: void 0
    }
  },
  setup(props, { slots }) {
    const styleObject = (0, import_vue.computed)(() => {
      if (typeof props.style === "string" || Array.isArray(props.style)) {
        return {};
      }
      return props.style;
    });
    return () => {
      var _a;
      return (0, import_vue.h)(
        props.as,
        {
          "style": {
            ...withMargin({
              m: props.m,
              mx: props.mx,
              my: props.my,
              mt: props.mt,
              mr: props.mr,
              mb: props.mb,
              ml: props.ml
            }),
            ...styleObject.value
          }
        },
        (_a = slots.default) == null ? void 0 : _a.call(slots)
      );
    };
  }
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Heading
});
